src/app/_components/FeeSection/index.tsx

```diff
@@ -0,0 +1,191 @@
+  return (
+    <Icon w="3.5" h="3.5" color="iconPrimary">
+      {children}
+    </Icon>
+  );
+}
+
+function FeeData({ title, ustxValue }: { title: ReactNode; ustxValue: number }) {
+  const stxValue = Number((ustxValue / MICROSTACKS_IN_STACKS).toFixed(2));
+  const formattedStxValue = stxValue < 0.01 ? '<0.01' : stxValue;
+  return (
+    <Stack
+      gap={2}
+      px={4}
+      pt={3}
+      pb={2}
+      borderRadius={'redesign.md'}
+      border="1px solid"
+      borderColor="redesignBorderSecondary"
+      flex="1"
+      minWidth="fit-content"
+    >
+      <Text textStyle="text-medium-lg" color="textPrimary">
+        {title}
+      </Text>
+      <HStack align="center" gap={1} whiteSpace={'nowrap'}>
+        <Text textStyle="text-mono-sm" color="textPrimary">
+          {formattedStxValue} STX
+        </Text>
+        <Text textStyle="text-mono-sm" color="textSecondary">
+          /
+        </Text>
+        <Text textStyle="text-mono-sm" color="textSecondary">
+          $0.01
+        </Text>
+      </HStack>
+    </Stack>
+  );
+}
+
+function FeeTabs() {
+  const { data: mempoolFeeResponse } = useMempoolFee();
+  const availableTxTypeFees = Object.keys(mempoolFeeResponse || {}).filter(
+    key => key !== 'all'
+  ) as MempoolFeeResponseKey[];
+  const avgFee = mempoolFeeResponse?.all;
+  return (
+    <TabsRoot variant={'primary'} size={'redesignMd'} defaultValue={'average'} gap="4">
+      <Stack gap={4} w="100%">
+        <Stack gap={3}>
+          <TabsList flexWrap={'wrap'} rowGap={2}>
+            <TabsTrigger value="average" gap={1}>
+              <TabIcon>
+                <PlusMinus weight="bold" />
+              </TabIcon>
+              <Text textStyle={'text-medium-sm'} color="textPrimary">
+                Average
+              </Text>
+            </TabsTrigger>
+            {availableTxTypeFees.map(txType => (
+              <TabsTrigger key={txType} value={txType} gap={1}>
+                <TabIcon>{getTxTypeIcon(txType)}</TabIcon>
+                <Text textStyle={'text-medium-sm'} color="textPrimary">
+                  {getTxTypeLabel(txType)}
+                </Text>
+              </TabsTrigger>
+            ))}
+          </TabsList>
+          <Flex>
+            <TabsContent value="average">
+              <HStack gap={2} flexWrap={'wrap'}>
+                <FeeData title="Low" ustxValue={avgFee?.low_priority || 0} />
+                <FeeData title="Standard" ustxValue={avgFee?.medium_priority || 0} />
+                <FeeData
+                  title={
+                    <>
+                      High{' '}
+                      <Icon w="4" h="4" color="iconTertiary">
+                        <Lightning weight="fill" />
+                      </Icon>
+                    </>
+                  }
+                  ustxValue={avgFee?.high_priority || 0}
+                />
+              </HStack>
+            </TabsContent>
+            {availableTxTypeFees.map(txType => (
```

Comment:
Why do we need to have a special case for all, which corresponds to the average tab here? I think you would just need to have a special case for all in your getTxTypeLabel function